controllers/openstackmachine_controller_test.go

```diff
@@ -89,8 +89,9 @@ func getDefaultOpenStackMachine() *infrav1.OpenStackMachine {
 			ServerMetadata: map[string]string{
 				"test-metadata": "test-value",
 			},
-			ConfigDrive:   pointer.Bool(true),
-			ServerGroupID: serverGroupUUID,
+			ConfigDrive:    pointer.Bool(true),
+			SecurityGroups: []infrav1.SecurityGroupFilter{},
```

Comment:
nit: I think you're setting SecurityGroups to an empty list to avoid appending to nil here? If so, you don't need to do that: nil is a valid empty slice. Appending to it will create a slice in the same way that a new slice is created if you exceed the capacity of the existing slice: https://go.dev/play/p/Nhln2sbvEGD

It's also safe and idiomatic to iterate over a nil slice in a for loop: https://go.dev/play/p/E5aqZHV51L0. This means, e.g. you don't need to write an additional guard like:
```go
if (slice != nil) {
    for _, elem := range slice {
        ...
    }
}
```

Reply:
Good to know, but the reason I was dumber: I has set `SecurityGroups` to an empty list in the test cases, and I modified the default machine to avoid having to also create a "custom" machine in the test case, which in turn meant I had to modify the default instance spec :sweat_smile: With the benefit of hindight, I could have simply not set `SecurityGroups` in those two additional test cases and thus avoided modifying anything in the defaults. lmk if you want me to respin.

Reply:
Not worth a respin just for this imho.